### ISSUE TITLE
twisted.trial.reporter's deprecated methods should be removed

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7815]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7815|
|Type|enhancement|
|Created|2015-03-16 09:13:34Z|
|Branch|https://github.com/twisted/twisted/tree/testresult-removedeprecated-7815|

It's been deprecated since 2008, so the Py3 porting time sounds like a good time to do it.

<details><summary>Searchable metadata</summary>

```
trac-id__7815 7815
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__branches_testresult_removedeprecated_7815 branches/testresult-removedeprecated-7815
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1426497214991601 1426497214991601
changetime__1426863623051211 1426863623051211
version__None None
owner__hawkowl hawkowl
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

(In [44055]) Branching to testresult-removedeprecated-7815.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Builders spun, plz reviewz :)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban set owner to @hawkowl|
|-|-|

not all builders are green ... pyflakes and twistechecker.

btw. you can not run pyflakes locally on the files that you have changed as the Twisted code is much cleaner of old pyflakes errors

---

Changes looks good.
Please clean the code and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44168]) Merge testresult-removedeprecated-7815: twisted.trial.reporter's deprecated methods should be removed

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7815](https://github.com/twisted/twisted/issues/7815)
